stan/math/rev/fun/multiply_log.hpp

```diff
@@ -105,6 +105,265 @@ inline var multiply_log(double a, const var& b) {
   return var(new internal::multiply_log_dv_vari(a, b.vi_));
 }
 
+/**
+ * Return the elementwise product `a * log(b)`.
+ *
+ * Both `T1` and `T2` are matrices, and one of `T1` or `T2` must be a
+ * `var_value`
+ *
+ * @tparam T1 Type of first argument
+ * @tparam T2 Type of second argument
+ * @param a First argument
+ * @param b Second argument
+ * @return elementwise product of `a` and `log(b)`
+ */
+template <typename T1, typename T2, require_all_matrix_t<T1, T2>* = nullptr,
+          require_any_var_matrix_t<T1, T2>* = nullptr>
+inline auto multiply_log(const T1& a, const T2& b) {
+  check_matching_dims("multiply_log", "a", a, "b", b);
+  if (!is_constant<T1>::value && !is_constant<T2>::value) {
+    arena_t<promote_scalar_t<var, T1>> arena_a = a;
+    arena_t<promote_scalar_t<var, T2>> arena_b = b;
+
+    return make_callback_var(
+        multiply_log(arena_a.val(), arena_b.val()),
+        [arena_a, arena_b](const auto& res) mutable {
+          for (Eigen::Index j = 0; j < res.adj().cols(); ++j) {
+            for (Eigen::Index i = 0; i < res.adj().rows(); ++i) {
+              if (unlikely(is_any_nan(arena_a.val().coeff(i, j),
+                                      arena_b.val().coeff(i, j)))) {
+                arena_a.adj().coeffRef(i, j) = NOT_A_NUMBER;
+                arena_b.adj().coeffRef(i, j) = NOT_A_NUMBER;
+              } else {
+                arena_a.adj().coeffRef(i, j)
+                    += res.adj().coeff(i, j) * log(arena_b.val().coeff(i, j));
+                if (arena_b.val().coeff(i, j) == 0.0
+                    && arena_a.val().coeff(i, j) == 0) {
+                  arena_b.adj().coeffRef(i, j) += res.adj().coeff(i, j) * INFTY;
```

Comment:
Is there a reason to mulitply res's adjoint by infinity here and not just return infty?

Reply:
I did a bunch of cleaning to multiply_log. I got rid of these infinity things in the process.

I don't think it makes sense to talk about the derivative of `a log(b)` with respect to b at `a == 0` and `b == 0`.

`a / b` doesn't exist at `a = 0`, `b = 0`, and if you try to compute a limit as `a -> 0 & b -> 0`, you get different things depending on the angle of approach (`a == b`, `b -> 0` gives a limit of 1 and `a == sqrt(b)`, `b -> 0` gives a limit of infinity), so I doubt this is even a thing. 

I got rid of the NaN checks too. I'm not sure what they're doing. It was part of this [change](https://github.com/stan-dev/math/commit/5306a5ccd66c4d3435c24ba0b41ed851fb4bea70), but I didn't find an issue or pull this was attached to.